hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientScheduledExecutorProxy.java

```diff
@@ -382,12 +394,19 @@ private int getTaskOrKeyPartitionId(Runnable task, Object key) {
     }
 
     private String extractNameOrGenerateOne(Object command) {
-        String name = null;
-        if (command instanceof NamedTask) {
-            name = ((NamedTask) command).getName();
-        }
+        String taskName = getNamedTaskName(command);
+        return taskName != null ? taskName : UuidUtil.newUnsecureUuidString();
+    }
 
-        return name != null ? name : UuidUtil.newUnsecureUuidString();
+    private String getNamedTaskName(Object command) {
+        if (command instanceof AbstractTaskDecorator) {
```

Comment:
I like what you did here, but unfortunately the `NamedTask` marker is public, meaning anyone can implement it directly without the use of our helper. In this case, the `AbstractTaskDecorator` inheritance expectation is not going to be valid.